packages/core/src/codewhisperer/region/regionProfileManager.ts

```diff
@@ -110,7 +139,56 @@ export class RegionProfileManager {
         if (conn === undefined || !isSsoConnection(conn)) {
             return []
         }
+
+        // WaitUntil lock is acquired, the API tps is low, server can't afford multiple IDE instances making the calls simultaneiously
+        const cached = await waitUntil(
```

Comment:
should this be encapsulated in `tryAcquireLock`? else the name `tryAcquireLock` is misleading, and confusing.

Reply:
sg